zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java

```diff
@@ -317,79 +322,81 @@ public CommandResponse run(ZooKeeperServer zkServer, Map<String, String> kwargs)
 
-            response.put("global_sessions", zkdb.getSessionCount());
-            response.put("local_sessions",
-                    zkServer.getSessionTracker().getLocalSessionCount());
+                response.put("watch_count", zkdb.getDataTree().getWatchCount());
+                response.put("ephemerals_count", zkdb.getDataTree().getEphemeralsCount());
+                response.put("approximate_data_size", zkdb.getDataTree().cachedApproximateDataSize());
 
-            OSMXBean osMbean = new OSMXBean();
-            response.put("open_file_descriptor_count", osMbean.getOpenFileDescriptorCount());
-            response.put("max_file_descriptor_count", osMbean.getMaxFileDescriptorCount());
-            response.put("connection_drop_probability", zkServer.getConnectionDropChance());
+                response.put("global_sessions", zkdb.getSessionCount());
+                response.put("local_sessions",
+                        zkServer.getSessionTracker().getLocalSessionCount());
 
-            response.put("last_client_response_size", stats.getClientResponseStats().getLastBufferSize());
-            response.put("max_client_response_size", stats.getClientResponseStats().getMaxBufferSize());
-            response.put("min_client_response_size", stats.getClientResponseStats().getMinBufferSize());
+                response.put("connection_drop_probability", zkServer.getConnectionDropChance());
 
-            if (zkServer instanceof QuorumZooKeeperServer) {
-                QuorumPeer peer = ((QuorumZooKeeperServer) zkServer).self;
-                response.put("quorum_size", peer.getQuorumSize());
-            }
+                response.put("last_client_response_size", stats.getClientResponseStats().getLastBufferSize());
+                response.put("max_client_response_size", stats.getClientResponseStats().getMaxBufferSize());
+                response.put("min_client_response_size", stats.getClientResponseStats().getMinBufferSize());
 
-            if (zkServer instanceof LeaderZooKeeperServer) {
-                Leader leader = ((LeaderZooKeeperServer) zkServer).getLeader();
-
-                response.put("learners", leader.getLearners().size());
-                response.put("synced_followers", leader.getForwardingFollowers().size());
-                response.put("synced_non_voting_followers", leader.getNonVotingFollowers().size());
-                response.put("synced_observers", leader.getObservingLearners().size());
-                response.put("pending_syncs", leader.getNumPendingSyncs());
-                response.put("leader_uptime", leader.getUptime());
-
-                response.put("last_proposal_size", leader.getProposalStats().getLastBufferSize());
-                response.put("max_proposal_size", leader.getProposalStats().getMaxBufferSize());
-                response.put("min_proposal_size", leader.getProposalStats().getMinBufferSize());
-            }
+                if (zkServer instanceof QuorumZooKeeperServer) {
```

Comment:
Ops.
I have changed this code on https://github.com/apache/zookeeper/pull/918
Maybe we can commit #918 as @enixon you already reviewed it and we are waiting for the final word from @lvfangmin

Reply:
:) 

Not an issue, I'm happy to fix up any conflicts if yours lands first.